Add "Walking Skeleton"

iaindillingham opened the pull request:
This defines *walking skeleton* and captures what I think Ethelred's walking skeleton should be. In several respects, Ethelred's walking skeleton is retrospective: we're already implementing it, even in its second iteration. However, I think it's good to write it down.

Because I'm still unsure about walking skeletons, I'm creating a draft PR and **asking** for a review.

benbc reviewed (commented):
This looks good to me.

Maybe "every day" is unnecessarily constraining. I'm not sure that you'd really get value from forcing yourselves to do the manual version every day. "On a schedule"?

iaindillingham committed:
Add "Walking Skeleton"

This defines *walking skeleton* and captures what I think Ethelred's
walking skeleton should be. In several respects, Ethelred's walking
skeleton is retrospective: we're already implementing it, even in it's
second iteration. However, I think it's good to write it down.
